sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamSetOperatorRelBase.java

```diff
@@ -63,16 +62,6 @@ public BeamSetOperatorRelBase(BeamRelNode beamRelNode, OpType opType, boolean al
         inputs);
     PCollection<Row> leftRows = inputs.get(0);
     PCollection<Row> rightRows = inputs.get(1);
-    Schema leftSchema = leftRows.getSchema();
-    Schema rightSchema = rightRows.getSchema();
-    if (!leftSchema.typesEqual(rightSchema)) {
-      throw new IllegalArgumentException(
-          "Can't intersect two tables with different schemas."
-              + "lhsSchema: "
-              + leftSchema
-              + "  rhsSchema: "
-              + rightSchema);
-    }
```

Comment:
Shouldn't we still verify that the schemas have equivalent types without nullability?

Reply:
The merge function itself now verifies this.

Reply:
Ah I see. If there aren't any tests verifying this already could you add one?

Reply:
There is a test in CoGroupTest.java that verifies this. Should I add an additional one here?

Reply:
It could be nice to have something in ZetaSQLDialectSpecTest to verify the set operation just in case we break the logic there and not in CoGroup, but I guess it's fine either way